tool/tsh/common/tsh.go

```diff
@@ -1908,6 +1913,8 @@ func onLogin(cf *CLIConf, reExecArgs ...string) error {
 		cf.DesiredRoles = ""
 	}
 
+	cf.disableHardwareKeyAgentClient = true
```

Comment:
I'd add a clarifying comment here.

```suggestion
	// For login operations, we use the hardware key
	// service directly instead of the agent.
	cf.disableHardwareKeyAgentClient = true
```